### ISSUE TITLE
[Gmail]/All Mail (Failure)

### ISSUE THREAD

**Issue by Marly212** [AnalogJ/hatchet#1]
Hey,

I am running into the following error while trying to use your program.

![grafik](https://user-images.githubusercontent.com/42452096/149681029-9636df70-5c39-435b-a7d2-492a99eac72d.png)

**Comment by AnalogJ** [comment]
Are you using a Gmail account or something else?

**Comment by AnalogJ** [comment]
Actually, this could also be caused by the "All Mail" inbox not being available in IMAP.
You can enable this by going to your Gmail settings, clicking "Labels" and checking the "Show in IMAP" checkbox beside "All Mail" 

<img width="1256" alt="Screen Shot 2022-01-16 at 4 30 17 PM" src="https://user-images.githubusercontent.com/891875/149684428-cc8b9873-e04a-4d7e-91c0-881a091b75a2.png">

**Comment by Marly212** [comment]
I am using a Gmail account, and the label "All Mail" is enabled and is shown in IMAP.

![grafik](https://user-images.githubusercontent.com/42452096/149724389-a844d735-66f8-461f-ade3-3fb3712580e0.png)

**Comment by ihmpavel** [comment]
Hi, I ran into same issue. 

You Gmail account has to be in English and `Show in IMAP` for `All Mail` should be enabled (it is enabled by default).

How to (temporarily) change Gmail language (to _English US_) 👉  https://support.google.com/mail/answer/17091

But it is very weird, that getting emails by system label is dependent on the user language...

**Comment by Marly212** [comment]
Hi, thanks that one fixed it. I agree its very weird

**Comment by ihmpavel** [comment]
Maybe @AnalogJ could mention it in the README, in the `Troubleshooting` section? 🤔 Both fixes (_IMAP enabled_ and _language_)?

**Comment by AnalogJ** [comment]
Interesting, I guess it makes sense that the label based mailboxes would be language specific. Would you be willing to take a screenshot of the system labels in your native languages? I want to do a bit of research & experimentation before I update the readme.

**Comment by ihmpavel** [comment]
Hi, I tried to get localized labels with official Google [API](https://developers.google.com/gmail/api/reference/rest/v1/users.labels/list?apix_params=%7B%22userId%22%3A%22me%22%7D), but it does not return localized names - only English ones.

I checked the [link in the source code](https://github.com/AnalogJ/hatchet/blob/main/pkg/engine.go#L77). Same information is on the official [Google Developers page](https://developers.google.com/gmail/imap/imap-extensions#special-use_extension_of_the_list_command)

IMAP support `listing` all folders, that is good for us.

In my language (`Czech` - čeština), label `All emails` in Gmail is translated as `Všechny zprávy` (in Gmail diacritics is encoded as `V&AWE-echny zpr&AOE-vy`)

I created small Python script, which lists all available folders ([Inspiration](https://stackoverflow.com/questions/44230855/python-imaplib-selecting-folders))
```
from imaplib import IMAP4_SSL

mail = IMAP4_SSL("imap.gmail.com")
mail.login(EMAIL, PASSWORD)

for i in mail.list()[1]:
    print(i)
```

Result 
![image](https://user-images.githubusercontent.com/42217494/149825524-a471f434-ae91-40c0-9e24-46cec5b22cc9.png)
☝️ check last line - it has `\All`. The script should be matching (via pattern) `\All`, instead of hardcoding `[Gmail]/All Mail` `https://github.com/AnalogJ/hatchet/blob/main/pkg/engine.go#L78`

Python `IMAP.list()` https://docs.python.org/3/library/imaplib.html#imaplib.IMAP4.list

    
Inspired by https://stackoverflow.com/a/22796586 (first comment), https://stackoverflow.com/questions/44230855/python-imaplib-selecting-folders

I have never worked on this low level with IMAP, but this is cool. Sorry for choosing `Python` for example, but I can at least print something to console, instead with `Go`

**Comment by AnalogJ** [comment]
FYI, the ability to specify the imap mailbox name is available in v0.0.4, via the `--imap-mailbox-name` CLI flag.

**Comment by AnalogJ** [comment]
thanks for your help @ihmpavel using your code as a guide I was able to use the `\All` attribute to correctly find the correct (language agnostic) mailbox name - https://github.com/AnalogJ/hatchet/commit/8e11bc4317aed2de32dd8d5a13ecb32dfe9a8205

This change is available in v0.0.5 and onwards

**Comment by ihmpavel** [comment]
Hi, I have just tried `v0.0.5` which is working like a charm. Good job! 🥳
